src/net.cpp

```diff
@@ -2780,19 +2780,19 @@ void CConnman::ThreadOpenAddedConnections()
     AssertLockNotHeld(m_unused_i2p_sessions_mutex);
     while (true)
     {
-        CSemaphoreGrant grant(*semAddnode);
```

Comment:
I think that this is a small change in behavior (fyi @vasild because of https://github.com/bitcoin/bitcoin/pull/28331#discussion_r1341652305):
On master, the thread will block here if all addnode slots are taken because we call `Acquire()` (and not `TryAcquire()`) so there won't be any `sleep_for` and no continuous calls to `GetAddedNodeInfo()` in this situation (until the grant is released because one of the existing added nodes has disconnected).
With this PR removing non-fTry acquiry of the grant, it means that we will now continuously cycle through the while loop, calling `GetAddedNodeInfo()` and sleeping for 2 seconds in the same situation where all addnode slots are taken.
Is this change intended / better than the behavior on master? On first glance, repeated calls to `GetAddedNodeInfo()` seem unnecessary when we don't have a grant anyway, so the behavior on master seems more natural.

Reply:
Looks good, I tested that the behavior doesn't change anymore.